plugin.rb

```diff
@@ -44,7 +44,14 @@ def toggle_noindex
 
       def show
         super
-        response.headers["X-Robots-Tag"] = "noindex" if @topic_view.topic.noindex
+        topic =
+          if @topic_view.nil? && params[:id]
+            Topic.find_by_slug(params[:id])
```

Comment:
I think it may be cleaner to just do nothing when the URL doesn't have a topic ID. In core, a request to `/t/slug` will redirect to `/t/slug/topic-id` and the noindex rule should apply after the redirect, right?

Reply:
Yeah, I've also realized that I was testing if it was adding a response header to a redirect page...

Thanks for the review!